Shared/Samples/Create and save map/CreateAndSaveMapView.swift

```diff
@@ -0,0 +1,419 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import ArcGISToolkit
+import SwiftUI
+
+/// A view that allows a user to create a map and save it to a portal.
+struct CreateAndSaveMapView: View {
+    /// The authenticator to handle authentication challenges.
+    @StateObject private var authenticator = Authenticator()
+    
+    /// The portal to save the map to.
+    @State private var portal = Portal(
+        url: URL(string: "https://www.arcgis.com")!,
+        connection: .authenticated
+    )
+    
+    /// The map that we will save to the portal.
+    @State private var map: Map?
+    
+    /// The error that occurred, if any, when trying to save the map to the portal.
+    @State private var error: Error?
+    
+    /// The status of the sample workflow.
+    @State private var status: Status = .loadingPortal
+    
+    /// The API key to use temporarily while using OAuth.
+    @State private var apiKey: APIKey?
+    
+    /// The portal user's folders that you can save the map to.
+    @State private var folders: [PortalFolder]?
```

Comment:
```suggestion
    @State private var folders: [PortalFolder] = []
```

It seems odd to me that this collection is optional. Is it needed for some reason?